temporalcli/commands.taskqueue.go

```diff
@@ -11,31 +11,237 @@ import (
+		}
+	}
+	return rRows, nil
+}
+
+func taskQueueTypeToStr(taskQueueType client.TaskQueueType) (string, error) {
+	switch taskQueueType {
+	case client.TaskQueueTypeUnspecified:
+		return "unspecified", nil
+	case client.TaskQueueTypeWorkflow:
+		return "workflow", nil
+	case client.TaskQueueTypeActivity:
+		return "activity", nil
+	case client.TaskQueueTypeNexus:
+		return "nexus", nil
+	default:
+		return "", fmt.Errorf("unrecognized task queue type: %d", taskQueueType)
+	}
+}
+
+func buildIDToPollerRows(pRows []pollerRowType, buildID string, typesInfo map[client.TaskQueueType]client.TaskQueueTypeInfo) ([]pollerRowType, error) {
+	for t, info := range typesInfo {
+		taskQueueType, err := taskQueueTypeToStr(t)
+		if err != nil {
+			return pRows, err
+		}
+		for _, p := range info.Pollers {
+			pRows = append(pRows, pollerRowType{
+				BuildID:        buildID,
+				TaskQueueType:  taskQueueType,
+				Identity:       p.Identity,
+				LastAccessTime: p.LastAccessTime,
+				RatePerSecond:  p.RatePerSecond,
+			})
+		}
+	}
+	return pRows, nil
+}
+
+func descriptionToPollerRows(taskQueueDescription client.TaskQueueDescription) ([]pollerRowType, error) {
+	var pRows []pollerRowType
+	var err error
+	// Unversioned queue first
+	val, ok := taskQueueDescription.VersionsInfo[client.UnversionedBuildID]
+	if ok {
+		pRows, err = buildIDToPollerRows(pRows, unversioned, val.TypesInfo)
+		if err != nil {
+			return nil, err
+		}
+	}
+	for k, val := range taskQueueDescription.VersionsInfo {
+		if k != client.UnversionedBuildID {
+			pRows, err = buildIDToPollerRows(pRows, k, val.TypesInfo)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return pRows, nil
+}
+
+func descriptionToRows(taskQueueDescription client.TaskQueueDescription) (taskQueueDescriptionType, error) {
+	rRows, err := descriptionToReachabilityRows(taskQueueDescription)
+	if err != nil {
+		return taskQueueDescriptionType{}, err
+	}
+	pRows, err := descriptionToPollerRows(taskQueueDescription)
+	if err != nil {
+		return taskQueueDescriptionType{}, err
+	}
+
+	return taskQueueDescriptionType{
+		Reachability: rRows,
+		Pollers:      pRows,
+	}, nil
+}
+
+func printTaskQueueDescription(cctx *CommandContext, taskQueueDescription client.TaskQueueDescription) error {
+	descRows, err := descriptionToRows(taskQueueDescription)
+	if err != nil {
+		return fmt.Errorf("creating task queue description rows failed: %w", err)
+	}
+
+	if !cctx.JSONOutput {
+		cctx.Printer.Println(color.MagentaString("Task Reachability:"))
+		err = cctx.Printer.PrintStructured(descRows.Reachability, printer.StructuredOptions{Table: &printer.TableOptions{}})
+		if err != nil {
+			return fmt.Errorf("displaying reachability failed: %w", err)
+		}
+
+		cctx.Printer.Println(color.MagentaString("Pollers:"))
+		return cctx.Printer.PrintStructured(descRows.Pollers, printer.StructuredOptions{Table: &printer.TableOptions{}})
+	}
+
+	// json output
+	return cctx.Printer.PrintStructured(descRows, printer.StructuredOptions{})
```

Comment:
I think it may be clearer to just print the proto describe response when in JSON mode. The printer is already prepared to properly print protos as JSON. No need for a bunch of custom JSON objects. And we usually find the protos are stable and easy to point users to if they want to see which fields would be present in their JSON.

Reply:
Talking with @ShahabT there are some things that we don't want to include, like `WorkerVersionCapabilities` that are mostly redundant, and sometimes expose Build IDs derived from code hashes that could be confusing. 
Also, I thought that a "row friendly" layout could be easier if they want to add results to a traditional DB, and more compatible with the text view...

Reply:
Our HTTP API is going to return that other JSON. But I don't mind if we want this separate JSON format for just this call, I have no strong opinion.